parts/k8s/cloud-init/artifacts/cse_main.sh

```diff
@@ -1,8 +1,13 @@
 #!/bin/bash
 ERR_FILE_WATCH_TIMEOUT=6 {{/* Timeout waiting for a file */}}
+ERR_ALREADY_RUN=116 {{/*Have already run and not sure if idempotent*/}
 
 set -x
 echo $(date),$(hostname), startcustomscript>>/opt/m
+if [ -s /opt/azure/containers/provision.complete ]; then
+      exit $ERR_ALREADY_RUN
```

Comment:
Do we actually want to return an error here, or is this just for gathering error rate data? As you've identified, the bootstrap scripts as-is are not implemented idempotently, so something unexpected causes them to re-run "accidentally", perhaps we should just return zero immediately.

Reply:
To me success is fine though it might be infuriating if you hit this path get a 0 and don't understand why. Ideally there's be multiple success code or I could just echo ALREADY RUN

Reply:
Let's move this exit statement to run before the datestamp echo statement